### PR TITLE
Added wrappers for shutil.copytree and os.chdir (WIP)

### PR DESCRIPTION
**Pull request by wpoely86**
To be used for the extended dry run.

### CODE REVIEW

File: easybuild/tools/filetools.py [hunk 1]
```diff
@@ -1003,80 +1006,9 @@ def cleanup(logfile, tempdir, testing, silent=False):
...
-        shutil.copystat(src, dst)
-    except OSError, why:
-        if WindowsError is not None and isinstance(why, WindowsError):
-            # Copying file access times may fail on Windows
-            pass
-        else:
-            errors.extend((src, dst, str(why)))
-    if errors:
-        raise Error(errors)
+    """Has been replaced by copy_files"""
```
Comment (boegel) [inline easybuild/tools/filetools.py]: include `DEPRECATED` in the docstring please

File: easybuild/tools/filetools.py [hunk 2]
```diff
@@ -1158,3 +1090,39 @@ def det_size(path):
         _log.warn("Could not determine install size: %s" % err)
 
     return installsize
+
+
+def copy_files(src, dst, symlinks=False, ignore=None):
+    """
+    Wrapper around shutil.copytree to nicely handle extended dry run
+    Arguments are the same as for the original function
+    The destination directoy must not exist
```
Comment (boegel) [inline easybuild/tools/filetools.py]: typo: `directory`

File: easybuild/tools/filetools.py [hunk 3]
```diff
@@ -1158,3 +1090,39 @@ def det_size(path):
...
+    """
+    if build_option('extended_dry_run'):
+        dry_run_msg("Copying directory %s to %s" % (src, dst), silent=build_option('silent'))
+        return
+
+    if os.path.exists(dst):
+        raise EasyBuildError("Destination directory %s of copy already exists.", dst)
```
Comment (boegel) [inline easybuild/tools/filetools.py]: why is this needed?

do we know why `copytree` does this?

it would be useful if we're able to cope with this...

File: easybuild/tools/filetools.py [hunk 4]
```diff
@@ -1158,3 +1090,39 @@ def det_size(path):
...
+
+
+def copy_files(src, dst, symlinks=False, ignore=None):
+    """
+    Wrapper around shutil.copytree to nicely handle extended dry run
+    Arguments are the same as for the original function
+    The destination directoy must not exist
+    """
+    if build_option('extended_dry_run'):
+        dry_run_msg("Copying directory %s to %s" % (src, dst), silent=build_option('silent'))
```
Comment (boegel) [inline easybuild/tools/filetools.py]: `copy_files` should be able to copy both files and directories imho...

File: easybuild/tools/filetools.py (hunk 3 above)
Comment (wpoely86) [inline easybuild/tools/filetools.py]: it's in the docs that it must be so. We could integrate it here to delete the directory if it exists but I'm not sure if that is what we want

File: easybuild/tools/filetools.py (hunk 4 above)
Comment (wpoely86) [inline easybuild/tools/filetools.py]: yes, it should. I will look if I find some time

File: easybuild/tools/filetools.py (hunk 3 above)
Comment (boegel) [inline easybuild/tools/filetools.py]: I don't see why we shouldn't, if the directory is empty, that is.
Otherwise the caller is simply going to do exactly that anyway?

File: easybuild/tools/filetools.py (hunk 4 above)
Comment (boegel) [inline easybuild/tools/filetools.py]: please make this `(WIP)` in the mean time

File: easybuild/tools/filetools.py (hunk 3 above)
Comment (wpoely86) [inline easybuild/tools/filetools.py]: well, it the dir is empty, sure

**Comment by wpoely86** [comment]
@boegel an attempt at including files, still needs proper testing

**Comment by boegel** [comment]
@wpoely86: keep me posted if you'v tested this through; are you up for adding tests for these too?

**Comment by boegel** [comment]
@wpoely86: please take into account https://github.com/hpcugent/easybuild-easyblocks/issues/492 here

File: easybuild/tools/filetools.py [hunk 5]
```diff
@@ -1158,3 +1102,61 @@ def det_size(path):
...
+                raise EasyBuildError("Failed to remove %s: %s", dst, err)
+
+    if isinstance(src, basestring):
+        src = [src]
+
+    try:
+        for item in src:
+            if os.path.isfile(item):
+                _log.debug("Copying file %s to %s" % (item, dst))
+                shutil.copy2(item, dst)
```
Comment (boegel) [inline easybuild/tools/filetools.py]: you should take into account that individual files can also be symlinks, since `copy2` resolves the symlink rather than copying it

cfr. http://stackoverflow.com/questions/4847615/copying-a-symbolic-link-in-python

**Comment by wpoely86** [comment]
This is going to stay WIP forever
